airbyte-server/src/main/java/io/airbyte/server/ConfigDumpImporter.java

```diff
@@ -387,4 +402,219 @@ private void checkDBVersion(final String airbyteVersion) throws IOException {
   }
 
+  public UploadRead uploadArchiveResource(File archive) {
+    try {
+      final UUID resourceId = UUID.randomUUID();
+      FileUtils.moveFile(archive, stagedResourceRoot.resolve(resourceId.toString()).toFile());
+      return new UploadRead()
+          .status(UploadRead.StatusEnum.SUCCEEDED)
+          .resourceId(resourceId);
+    } catch (IOException e) {
+      LOGGER.error("Failed to upload archive resource", e);
+      return new UploadRead().status(UploadRead.StatusEnum.FAILED);
+    }
+  }
+
+  public File getArchiveResource(UUID resourceId) {
+    final File archive = stagedResourceRoot.resolve(resourceId.toString()).toFile();
+    if (!archive.exists()) {
+      throw new IdNotFoundKnownException("Archive Resource not found", resourceId.toString());
+    }
+    return archive;
+  }
+
+  public void deleteArchiveResource(UUID resourceId) {
+    final File archive = getArchiveResource(resourceId);
+    FileUtils.deleteQuietly(archive);
+  }
+
+  public void importIntoWorkspace(String targetVersion, UUID workspaceId, File archive)
+      throws IOException, JsonValidationException, ConfigNotFoundException {
+    final Path sourceRoot = Files.createTempDirectory(Path.of("/tmp"), "airbyte_archive");
+    try {
+      // 1. Unzip source
+      Archives.extractArchive(archive.toPath(), sourceRoot);
+
+      // TODO: Auto-migrate archive?
+
+      // 2. dry run
+      try {
+        checkImport(targetVersion, sourceRoot);
+        importConfigsIntoWorkspace(sourceRoot, workspaceId, true);
+      } catch (Exception e) {
+        LOGGER.error("Dry run failed.", e);
+        throw e;
+      }
+
+      // 3. import configs
+      importConfigsIntoWorkspace(sourceRoot, workspaceId, false);
+    } finally {
+      FileUtils.deleteDirectory(sourceRoot.toFile());
+    }
+  }
+
+  private <T> void importConfigsIntoWorkspace(Path sourceRoot, UUID workspaceId, boolean dryRun)
+      throws IOException, JsonValidationException, ConfigNotFoundException {
+    // Keep maps of any re-assigned ids
+    final Map<UUID, UUID> sourceIdMap = new HashMap<>();
+    final Map<UUID, UUID> destinationIdMap = new HashMap<>();
+    final Map<UUID, UUID> connectionIdMap = new HashMap<>();
+    final Map<UUID, UUID> operationIdMap = new HashMap<>();
+
+    final List<String> directories = listDirectories(sourceRoot);
+    Stream<T> standardSyncs = null;
+
+    for (final String directory : directories) {
+      final Optional<ConfigSchema> configSchemaOptional = Enums.toEnum(directory.replace(".yaml", ""), ConfigSchema.class);
+
+      if (configSchemaOptional.isEmpty()) {
+        continue;
+      }
+      final ConfigSchema configSchema = configSchemaOptional.get();
+      final Stream<T> configs = readConfigsFromArchive(sourceRoot, configSchema);
+      if (!dryRun) {
+        switch (configSchema) {
+          case STANDARD_SOURCE_DEFINITION -> importStandardSourceDefinitionsIntoWorkspace(configs);
```

Comment:
How are we going to handle this on cloud now?

Reply:
I still need to introduce a wrapped version of this that changes the behavior in cloud

Reply:
For the moment, i'm thinking about adding a different configDumpImporter that ignores all standard definitions for  the short term (since we don't allow custom connectors yet)